### DISCUSSION TITLE
The problem in component update by changing the props

### DISCUSSION THREAD

**Discussion by ezattelli** [prabhuignoto/react-chrono#492] [General]
Hi, i have a problem with the component update when changing the props in react.
The setState command re-renders the component only for **_items_** and **_mode_**, and for the rest, I have to refresh the browser.

```
<Chrono
                    items={(Object.values((this.state.items)))}
                    mode={(((this.state.mode)))}

                    allowDynamicUpdate={this.state.allowDynamicUpdate}
                    enableLayoutSwitch={this.state.enableLayoutSwitch}
                    enableQuickJump={this.state.enableQuickJump}
                    showAllCardsHorizontal={this.state.showAllCardsHorizontal}
                    itemWidth={this.state.itemWidth}
                    timelinePointShape={this.state.timelinePointShape}
                    timelinePointDimension={this.state.timelinePointDimension}
                    cardHeight={this.state.cardHeight}
                    cardWidth={this.state.cardWidth}
                    scrollable={{scrollbar: this.state.scrollable}}
                    textOverlay={this.state.textOverlay}
                    slideShow={this.state.slideShow}
                    slideItemDuration={this.state.slideItemDuration}
                    slideShowType={this.state.slideShowType}
                    theme={(this.state.theme)}
                    fontSizes={(this.state.fontSizes)}
                    buttonTexts={(this.state.buttonTexts)}
                    enableDarkToggle={this.state.enableDarkToggle}
                    highlightCardsOnHover={this.state.highlightCardsOnHover}
                />
```

**Comment by prabhuignoto** [comment]
@ezattelli not all props are reactive. what prop are you trying to change. can you please mention those and why do you need all the props to be reactive?

**Reply by ezattelli** [reply]
Thanks for the reply, I'm making an app where users can choose options that depend on their needs, and maybe not all props are necessarily needed by everyone, but it's great to have everything.

**Reply by prabhuignoto** [reply]
@ezattelli thanks for your response. i am still curious and interested in knowing the exact use case for such a requirement. can you please elaborate the exact use case and what you are trying to achieve.

**Reply by ezattelli** [reply]
Currently working on Automation [Ignition](https://inductiveautomation.com/) and need a historical data visualization module. Found Chorono is a good choice for integration.
